File: T-Bank/Algorithms-and-Data-Structures-2025/1sprint-IntroductionToSorting-BinarySearch/contest/G-Anti-qsort_test/solution.py
import sys


def solution() -> None:
    """
    Time Complexity O(N)
    Memory Complexity O(N)
    """
    n: int = int(sys.stdin.readline().rstrip())
    nums: list[int] = list(range(1, n + 1))
    for idx in range(2, n):
        nums[idx], nums[idx // 2] = nums[idx // 2], nums[idx]
    sys.stdout.write(' '.join(str(num) for num in nums))
    sys.stdout.write('\n')


if __name__ == '__main__':
    solution()

File: T-Bank/Algorithms-and-Data-Structures-2025/1sprint-IntroductionToSorting-BinarySearch/contest/G-Anti-qsort_test/solution.cpp
#include <iostream>
#include <vector>
#include <algorithm>


void solution() {
    /*
    Time Complexity O(N)
    Memory Complexity O(N)
    */
    int n;
    std::cin >> n;
    std::vector<int> nums (n, 0);
    for (int i = 1; i <= n; ++i) nums[i - 1] = i;
    for (int idx = 2; idx < n; ++idx) {
        std::swap(nums[idx], nums[idx / 2]);
    }
    for (int i = 0; i < n; ++i) {
        if (i) std::cout << " ";
        std::cout << nums[i];
    }
    std::cout << "\n";
}


int main() {
    solution();
}